core/chains/evm/client/simulated_backend.go

```diff
@@ -0,0 +1,386 @@
+      {
+        "name": "_owner",
+        "type": "address"
+      }
+    ],
+    "name": "balanceOf",
+    "outputs": [
+      {
+        "name": "balance",
+        "type": "uint256"
+      }
+    ],
+    "payable": false,
+    "stateMutability": "view",
+    "type": "function"
+  }
+]`
+
+var balanceOfABI abi.ABI
+
+func init() {
+	var err error
+	balanceOfABI, err = abi.JSON(strings.NewReader(balanceOfABIString))
+	if err != nil {
+		panic(errors.Wrapf(err, "while parsing erc20ABI"))
+	}
+}
+
+// GetERC20Balance returns the balance of the given address for the token
+// contract address.
+func (c *SimulatedBackendClient) GetERC20Balance(address common.Address, contractAddress common.Address) (balance *big.Int, err error) {
+	callData, err := balanceOfABI.Pack("balanceOf", address)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while seeking the ERC20 balance of %s on %s",
+			address, contractAddress)
+	}
+	b, err := c.b.CallContract(context.Background(), ethereum.CallMsg{
+		To: &contractAddress, Data: callData},
+		c.currentBlockNumber())
+	if err != nil {
+		return nil, errors.Wrapf(err, "while calling ERC20 balanceOf method on %s "+
+			"for balance of %s", contractAddress, address)
+	}
+	err = balanceOfABI.UnpackIntoInterface(balance, "balanceOf", b)
+	if err != nil {
+		return nil, errors.New("unable to unpack balance")
+	}
+	return balance, nil
+}
+
+func (c *SimulatedBackendClient) GetLINKBalance(linkAddress common.Address, address common.Address) (*assets.Link, error) {
+	panic("not implemented")
+}
+
+// TransactionReceipt returns the transaction receipt for the given transaction hash.
+func (c *SimulatedBackendClient) TransactionReceipt(ctx context.Context, receipt common.Hash) (*types.Receipt, error) {
+	return c.b.TransactionReceipt(ctx, receipt)
+}
+
+func (c *SimulatedBackendClient) blockNumber(number interface{}) (blockNumber *big.Int, err error) {
+	switch n := number.(type) {
+	case string:
+		switch n {
+		case "latest":
+			return c.currentBlockNumber(), nil
+		case "earliest":
+			return big.NewInt(0), nil
+		case "pending":
+			panic("not implemented") // I don't understand the semantics of this.
+			// return big.NewInt(0).Add(c.currentBlockNumber(), big.NewInt(1)), nil
+		default:
+			blockNumber, err = utils.HexToUint256(n)
+			if err != nil {
+				return nil, errors.Wrapf(err, "while parsing '%s' as hex-encoded"+
+					"block number", n)
+			}
+			return blockNumber, nil
+		}
+	case *big.Int:
+		if n.Sign() < 0 {
+			return nil, fmt.Errorf("block number must be non-negative")
+		}
+		return n, nil
+	}
+	panic("can never reach here")
+}
+
+func (c *SimulatedBackendClient) HeadByNumber(ctx context.Context, n *big.Int) (*evmtypes.Head, error) {
+	if n == nil {
+		n = c.currentBlockNumber()
+	}
+	header, err := c.b.HeaderByNumber(ctx, n)
+	if err != nil {
+		return nil, err
+	} else if header == nil {
+		return nil, ethereum.NotFound
+	}
+	return &evmtypes.Head{
+		EVMChainID: utils.NewBigI(c.chainId.Int64()),
```

Comment:
this is the only change - return the specified chainID here to permit simulating different EVM chains all backed by the simulated backend (e.g. for CCIP integration tests)